sopel/config/core_section.py

```diff
@@ -949,11 +949,9 @@ def homedir(self):
 
     """
 
-    modes = ValidatedAttribute('modes', default='B')
+    modes = ValidatedAttribute('modes')
     """User modes to be set on connection.
 
-    :default: ``B``
-
```

Comment:
This should come with a `versionchanged` that explains that this used to be `B` by default.

Reply:
Have done. Take a look at the fixup/added commits; note I tagged them all with `[skip ci]` so we don't hammer GHA while the Coveralls reporting is broken. (I'll obviously have to do a rebase later, anyway, so it's fine to skip CI for now.)